### PR TITLE
Update docs for related to PointerCancelEvent to explain better where this is used

### PR DESCRIPTION
**Pull request by gslender**
In response to issue #93683 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [test-exempt] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/gestures/events.dart [hunk 1]
```diff
@@ -2036,7 +2036,9 @@ mixin _CopyPointerCancelEvent on PointerEvent {
   }
 }
 
-/// The input from the pointer is no longer directed towards this receiver.
+/// The input from the pointer is no longer directed towards this receiver. The
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/gestures/events.dart:2039]: According to the guide, the doc should starts with a short sentence. 
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#dartdoc-specific-requirements
Try to rephrase what you want to add in a second paragraph. 

Also, the sentence 'The "receiver" in this context does not refer to a widget, but the entire client' sounds kind of awkward out of context. What is this intended to explain or avoid?

**Review by dkwingsmt — CHANGES_REQUESTED** [review]

**Comment by gslender** [comment]
I'm sorry but not keen on proceeding with minor wording changes when I'm trying to fix ambiguity in the original docs. If someone closer to Flutter can't take ownership then I'm fine to have this dropped... I think the docs are poor, and so do you, so I attempted to fix them. If you feel they need more words/tweaks then please add/change what you want as I'm less motivated to fix what was already poorly worded - my changes removed that ambiguity and if you want it to be better than feel free to improve further without making it too verbose.

**Comment by dkwingsmt** [comment]
I understand. Thanks for your attempt.

**Comment by goderbauer** [comment]
I am closing this until we find somebody who can pick it up again.

**Comment by gslender** [comment]
That's a poor way to encourage community contributions. I make the effort of writing up what I felt explained the situation correctly, it was criticised with no example of what to write/document instead, and now closed and not fixed at all. All you've done is educated someone to not bother making the effort to contribute to what was already poorly worded and lacking.

**Comment by dkwingsmt** [comment]
I understand the frustration. I do appreciate every contribution, whether it's big or small, and the author experienced or first time. We merge many doc changes from all kinds of contributors every day.

But it doesn't mean that every PR can be merged without modification. We need to keep a certain style guide, both for quality and consistency throughout the repo. But you said you were not willing to make the changes I requested to make. So we can't merge it.

**Comment by gslender** [comment]
That's not true. You did not make any suggestions. You just stated you didn't like what was contributed. Anyway, won't debate further. Obviously you'd rather have misleading and incorrect docs than any improvements to fix that at all ??? Odd and certainly a strange way to improve Flutter. Best of luck.

**Comment by Sistem-Pack** [comment]
Hi, everyone! Maybe I have a simillar problem. If I touch the AnimatedHelperEx and lead with my finger vertically on the screen opacity animation effect is not finish. I have this code: 
1. My custom animation opacity widget
`class AnimatedHelperEx extends StatefulWidget {
  final Widget? child;
  final GestureTapCallback? onPressed;
  final GestureTapCallback? onCancel;
  final bool useOpacity;
  final int? durationValue;
  final HitTestBehavior? behavior;

  const AnimatedHelperEx(
      {Key? key,
      required this.child,
      this.onPressed,
      this.useOpacity = true,
      this.durationValue = 350,
      this.behavior,
      this.onCancel})
      : super(key: key);

  @override
  State<AnimatedHelperEx> createState() => _AnimatedHelperStateEx();
}

class _AnimatedHelperStateEx extends State<AnimatedHelperEx> {
  double _opacity = 1.0;
  int _duration = 350;

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onVerticalDragUpdate: (DragUpdateDetails details) {},
      onTap: this.widget.onPressed,
      onTapCancel: () {
        setState(() {
          _duration = this.widget.durationValue!;
          _opacity = 1.0;
          widget.onCancel;
        });
      },
      child: Listener(
        onPointerDown: (PointerDownEvent event) {
          setState(() {
            _duration = 0;
            _opacity = 0.2;
          });
        },
        onPointerUp: (PointerUpEvent event) {
          setState(() {
            _duration = this.widget.durationValue!;
            _opacity = 1.0;
          });
        },
        onPointerCancel: (PointerCancelEvent event) {
          setState(() {
            _duration = widget.durationValue!;
            _opacity = 1.0;
          });
        },
        child: AnimatedOpacity(
          opacity: widget.useOpacity ? _opacity : 1,
          duration: widget.useOpacity
              ? Duration(milliseconds: _duration)
              : Duration(seconds: 0),
          child: this.widget.child,
        ),
      ),
    );
  }
}`
2. My bottom panel widget.
`AnimatedHelperEx(
                                  onPressed: () {
                                    showModalBottomSheet(
                                        backgroundColor: Colors.transparent,
                                        context: context,
                                        builder: (BuildContext context) {
                                          return StreamBuilder(
                                              stream: controller.stream,
                                              builder: (context, snapshot) {
                                                return PurseViewModalBottomSheet(
                                                  controller: controller,
                                                );
                                              });
                                        });
                                  },`

3. Addition to the 2 step. 
`class PurseViewModalBottomSheet extends StatefulWidget {
  final StreamController<double> controller;
  const PurseViewModalBottomSheet({Key? key, required this.controller})
      : super(key: key);

  @override
  State<PurseViewModalBottomSheet> createState() =>
      _PurseViewModalBottomSheetState();
}

class _PurseViewModalBottomSheetState extends State<PurseViewModalBottomSheet> {
  @override
  double position = 140;
  double globalPositionOnUpdate = 0;
  Widget build(BuildContext context) {
    return GestureDetector(
      onVerticalDragCancel: () {
        setState(() {
          position = 140;
        });
      },
      onVerticalDragEnd: (DragEndDetails details) {
        if (((screenPortraitSizeHeight - globalPositionOnUpdate) + position) <
            70) {
          print(screenPortraitSizeHeight - globalPositionOnUpdate);
          Navigator.pop(context);
        } else {
          setState(() {
            while (position > 141) {
              position = position - 1;
              widget.controller.add(position);
            }
            position = 140;
            widget.controller.add(position);
          });
        }
      },
      onVerticalDragUpdate: (DragUpdateDetails details) {
        setState(() {
          if (globalPositionOnUpdate == 0) {
            globalPositionOnUpdate = details.globalPosition.dy;
          } else {
            if (globalPositionOnUpdate > details.globalPosition.dy) {
              position = position + 0.5;
            } else {
              position = position - 0.5;
            }
          }
          /*position = screenPortraitSizeHeight - details.globalPosition.dy < 140
              ? position - 0.5
              : position + 0.5;*/
          widget.controller.add(position);
          globalPositionOnUpdate = details.globalPosition.dy;
        });
      },
      child: BackdropFilter(
        filter: ImageFilter.blur(sigmaX: 10, sigmaY: 10),
        child: ClipRRect(
          borderRadius: BorderRadius.only(
              topLeft: Radius.circular(20), topRight: Radius.circular(20)),
          child: Container(
            height: position,
            color: liteBlueColor,
            child: SingleChildScrollView(
              child: Column(
                  crossAxisAlignment: CrossAxisAlignment.center,
                  children: [
                    Center(
                      child: Padding(
                        padding: const EdgeInsets.only(top: 10, bottom: 5),
                        child: Container(
                          height: 4,
                          width: 45,
                          decoration: BoxDecoration(
                              color: Colors.white54,
                              borderRadius: BorderRadius.circular(20)),
                        ),
                      ),
                    ),
                    Center(
                      child: Padding(
                        padding: EdgeInsets.only(
                            top: 5, bottom: 5, left: 20, right: 20),
                        child: AnimatedHelperEx(
                          onPressed: () {
                            Navigator.push(
                                context,
                                PageTransition(
                                    child: IncomMoneyAccount(
                                      OpenType: 0,
                                    ),
                                    type: PageTransitionType.rightToLeft));
                          },
                          child: Container(
                            height: 50,
                            decoration: BoxDecoration(
                                color: darkBlueColor,
                                borderRadius: BorderRadius.circular(5)),
                            child: Padding(
                              padding: const EdgeInsets.only(left: 10),
                              child: Row(
                                children: [
                                  Icon(
                                    Icons.add_card,
                                    color: Colors.white54,
                                  ),
                                  Padding(
                                    padding: const EdgeInsets.only(left: 10),
                                    child: Text(
                                      'Bank Card',
                                      style: TextStyle(fontSize: 14, color: bb),
                                    ),
                                  )
                                ],
                              ),
                            ),
                          ),
                        ),
                      ),
                    ),
                    Center(
                      child: Padding(
                        padding: EdgeInsets.only(
                            top: 5, bottom: 5, left: 20, right: 20),
                        child: AnimatedHelperEx(
                          onPressed: () {
                            Navigator.push(
                                context,
                                PageTransition(
                                    child: IncomMoneyAccount(
                                      OpenType: 0,
                                    ),
                                    type: PageTransitionType.rightToLeft));
                          },
                          child: Container(
                            height: 50,
                            decoration: BoxDecoration(
                                color: darkBlueColor,
                                borderRadius: BorderRadius.circular(5)),
                            child: Padding(
                              padding: const EdgeInsets.only(left: 10),
                              child: Row(
                                children: [
                                  Container(
                                    height: 25,
                                    width: 25,
                                    child: Stack(
                                      alignment: Alignment.center,
                                      children: [
                                        Align(
                                          alignment: Alignment.centerLeft,
                                          child: Container(
                                            child: Icon(
                                              Icons.phone_android_sharp,
                                              color: Colors.white54,
                                            ),
                                            decoration: const BoxDecoration(
                                              color: darkBlueColor,
                                              shape: BoxShape.circle,
                                            ),
                                          ),
                                        ),
                                        Align(
                                          alignment: Alignment.bottomRight,
                                          child: Container(
                                            child: Text(
                                              String.fromCharCode(
                                                  Icons.add_sharp.codePoint),
                                              style: TextStyle(
                                                inherit: false,
                                                color: Colors.white54,
                                                fontSize: 14.0,
                                                fontWeight: FontWeight.w900,
                                                fontFamily:
                                                    Icons.add_sharp.fontFamily,
                                                package:
                                                    Icons.add_sharp.fontPackage,
                                              ),
                                            ),
                                            decoration: BoxDecoration(
                                              color: darkBlueColor,
                                              shape: BoxShape.circle,
                                            ),
                                          ),
                                        ),
                                      ],
                                    ),
                                  ),
                                  Padding(
                                    padding: const EdgeInsets.only(left: 10),
                                    child: Text(
                                      'Mobile phone',
                                      style: TextStyle(fontSize: 14, color: bb),
                                    ),
                                  )
                                ],
                              ),
                            ),
                          ),
                        ),
                      ),
                    ),
                  ]),
            ),
          ),
        ),
      ),
    );
  }
}`

**Comment by Sistem-Pack** [comment]
I'm apologize! I find solution [https://stackoverflow.com/questions/62749821/flutter-how-to-cancel-pointer-up-event](url).

**Pull request closed without merging**